src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/UseDefaultDllImportSearchPathsAttribute.cs

```diff
@@ -79,21 +79,25 @@ public override void Initialize(AnalysisContext context)
                     defaultValue: UnsafeBits);
                 var defaultDllImportSearchPathsAttributeOnAssembly = compilation.Assembly.GetAttributes().FirstOrDefault(o => o.AttributeClass.Equals(defaultDllImportSearchPathsAttributeTypeSymbol));
 
+                // Does not analyze local functions. To analyze local functions, we'll need to use RegisterSyntaxAction.
                 compilationStartAnalysisContext.RegisterSymbolAction(symbolAnalysisContext =>
                 {
-                    var symbol = symbolAnalysisContext.Symbol;
+                    var symbol = (IMethodSymbol)symbolAnalysisContext.Symbol;
 
-                    if (!symbol.IsExtern || !symbol.IsStatic)
+                    // LibraryImport methods will be partial, and the generated implementation will have a non-null PartialDefinitionPart property
+                    if (!symbol.IsExtern || !symbol.IsStatic || symbol.PartialDefinitionPart != null)
```

Comment:
```suggestion
                    if (!symbol.IsStatic || (!symbol.IsExtern && symbol.PartialDefinitionPart != null))
```

`LibraryImport` doesn't use `extern`. I think the above is what you want.

Reply:
I think actually I want to get of the IsExtern check all together since some generated code won't have it at all, but we'll still want to warn if the search paths aren't present.